sql/src/main/java/io/crate/analyze/VersionVisitor.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Licensed to CRATE Technology GmbH ("Crate") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.analyze;
+
+import com.google.common.collect.ImmutableSet;
+import io.crate.metadata.ColumnIdent;
+import io.crate.operation.operator.AndOperator;
+import io.crate.operation.operator.EqOperator;
+import io.crate.operation.operator.OrOperator;
+import io.crate.planner.symbol.*;
+
+/**
+ * Visitor which traverses functions to find _version columns and rewrites
+ * the function to a `True` Literal.
+ */
+public class VersionVisitor extends SymbolVisitor<Void, Symbol> {
```

Comment:
maybe call it `VersionRewriter` or something. I would also hide the SymbolVisitor interface (by using an inner class) an just expose a `rewrite (Function query)` or something like that.